File: src/backend/SellerLite.Domain/Entities/Order.cs
using SellerLite.Domain.Common;
using SellerLite.Domain.Entities.Enums;
using SellerLite.Domain.Entities.ValueObjects;

namespace SellerLite.Domain.Entities;

public class Order : BaseAuditableEntity
{
    public string OrderNumber { get; set; } = string.Empty;
    public OrderStatus Status { get; private set; } = OrderStatus.Pending;
    public PaymentMethod PaymentMethod { get; set; } = PaymentMethod.COD;
    public decimal TotalPrice { get; set; }
    public decimal ShippingFee { get; set; }
    public string? CustomerName { get; set; }
    public PhoneNumber? CustomerPhone { get; set; }
    public List<OrderItem> Items { get; set; } = new();
    public void UpdateStatus(OrderStatus newStatus)
    {
        if (Status == OrderStatus.Cancelled || Status == OrderStatus.Returned || Status == OrderStatus.Completed)
        {
            return;
        }
        Status = newStatus;
    }
}

File: src/frontend/src/types/order.ts
import { OrderStatus } from '@/components/orders/OrderStatusBadge';

export interface Order {
  id: string;
  orderNumber: string;
  status: OrderStatus | string;
  totalPrice: number;
  shippingFee: number;
  customerName: string;
  customerPhone?: string;
  createdAt: string;
}

export interface OrderItem {
  productId: string;
  productName: string;
  sku: string;
  quantity: number;
  unitPrice: number;
}

export interface OrderDetail extends Order {
  items: OrderItem[];
  shippingAddress?: string;
  note?: string;
}

export interface CreateOrderRequest {
  customerName: string;
  customerPhone: string;
  items: {
    productId: string;
    quantity: number;
  }[];
}
